### ISSUE TITLE
[dev] API for combat simulation results

### ISSUE THREAD

**Issue by CeH9** [HearthSim/Hearthstone-Deck-Tracker#4317]
Would be nice to have API for combat simulation results:

``` kotlin
float PlayerLethalRate 
float WinRate 
float TieRate 
float LossRate 
float OpponentLethalRate 

int Iterations
```

It's useful in general, but my usecase is to detect opponent's high roll when player lose match.
e.g. if (opponent_lethal_rate < X) {...}

IIRC it's calculated right before combat begins (**Opponent's TurnStart**), if i'm wrong than would be nice to have additional event: **OnCombatSimulationReady(...)**

**Comment by sgkoishi** [comment]
https://github.com/HearthSim/Hearthstone-Deck-Tracker/blob/0bd5bac9f62bfb8ec94e6eaa650467572b9d874b/Hearthstone%20Deck%20Tracker/BobsBuddy/BobsBuddyInvoker.cs#L445-L456

All those methods, classes and the BobsBuddy library are either `private`, `internal` or source unavailable - I'm not sure if that means something. You may still find some way to patch the existing method :)

**Comment by KkEeVvIiNnn** [comment]
have u already solved this problem?
I am faced with the same problem ( want to get the win/tie/loss rate after each turn simulation done ) for developing an analyze plugin
thanks a lot!!

**Comment by KkEeVvIiNnn** [comment]
> > have u already solved this problem?
> > I am faced with the same problem ( want to get the win/tie/loss rate after each turn simulation done ) for developing an analyze plugin
> > thanks a lot!!
> 
> Both the BobsBuddy library and all related methods and classes are either `private`, `internal` or source unavailable. I would guess they are intended to keep those data secretly without public access.
> 
> ~You could still try to 'patch' them using tools like [Harmony](https://github.com/pardeike/Harmony/)~

so does it means that i need to run my own simulation? ( can't just fetch the results that HDT already calced )

**Comment by sgkoishi** [comment]
> 
> 
> > > have u already solved this problem?
> > > I am faced with the same problem ( want to get the win/tie/loss rate after each turn simulation done ) for developing an analyze plugin
> > > thanks a lot!!
> > 
> > 
> > Both the BobsBuddy library and all related methods and classes are either `private`, `internal` or source unavailable. I would guess they are intended to keep those data secretly without public access.
> > ~You could still try to 'patch' them using tools like [Harmony](https://github.com/pardeike/Harmony/)~
> 
> so does it means that i need to run my own simulation? ( can't just fetch the results that HDT already calced )

By patching you can change the logic of HDT itself, thus 'steal' those results once the simulation is done.
Without patching you could still get the result from `Hearthstone_Deck_Tracker.BobsBuddy.BobsBuddyInvoker.Output` using reflection, but there's no way to know when the simulation is done.

**Comment by KkEeVvIiNnn** [comment]
> > > > have u already solved this problem?
> > > > I am faced with the same problem ( want to get the win/tie/loss rate after each turn simulation done ) for developing an analyze plugin
> > > > thanks a lot!!
> > > 
> > > 
> > > Both the BobsBuddy library and all related methods and classes are either `private`, `internal` or source unavailable. I would guess they are intended to keep those data secretly without public access.
> > > ~You could still try to 'patch' them using tools like [Harmony](https://github.com/pardeike/Harmony/)~
> > 
> > 
> > so does it means that i need to run my own simulation? ( can't just fetch the results that HDT already calced )
> 
> By patching you can change the logic of HDT itself, thus 'steal' those results once the simulation is done.
> Without patching you could still get the result from `Hearthstone_Deck_Tracker.BobsBuddy.BobsBuddyInvoker.Output` using reflection, but there's no way to know when the simulation is done.

Thank you very much. I have to learn more about what you've mentioned.😅
